en/lessons/advanced/nerves.md

```diff
@@ -0,0 +1,262 @@
+
+Nerves environment
+  MIX_TARGET:   rpi3
+  MIX_ENV:      dev
+Resolving Nerves artifacts...
+  Resolving nerves_system_rpi3
+  => Trying https://github.com/nerves-project/nerves_system_rpi3/releases/download/v1.7.0/nerves_system_rpi3-portable-1.7.0-17EA89A.tar.gz
+|==================================================| 100% (133 / 133) MB
+  => Success
+  Resolving nerves_toolchain_arm_unknown_linux_gnueabihf
+  => Trying https://github.com/nerves-project/toolchains/releases/download/v1.1.0/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.1.0-2305AD8.tar.xz
+|==================================================| 100% (50 / 50) MB
+  => Success
+```
+
+Note that you need to set up an environment variable before running `mix deps.get`, as it downloads the system image and toolchain for the provided platform.
+
+## Burning the firmware
+
+Now we can proceed to actual flashing of the drive. Put it into the card reader, and if you set up everything correctly in previous steps, after running `mix firmware.burn` and confirming the device to use you should get this prompt:
+
+```
+Building ......../network_led/_build/rpi_dev/nerves/images/network_led.fw...
+Use 7.42 GiB memory card found at /dev/rdisk2? [Yn]
+```
+
+If you are sure this is the card you want to burn - pick `Y` and after some time the memory card is ready:
+
+```
+Use 7.42 GiB memory card found at /dev/rdisk2? [Yn]
+|====================================| 100% (32.51 / 32.51) MB
+Success!
+Elapsed time: 8.022 s
+```
+
+Now it's time to put the memory card into your hardware platform of choice and verify whether it works.
+
+If you have a screen connected - you should see a Linux boot sequence on it after powering up the device with this memory card inserted.
+
+## Setting up networking
+
+Next step - if we want a networked led device - is setting up a network. Nerves ecosystem has some exciting packages, and [nerves_network](https://github.com/nerves-project/nerves_network) is one of them.
+
+It's already present in your project - as a dependency of `nerves_init_gadget`. However, by default, it uses DHCP (see config for it in `config/config.exs` after the `config :nerves_init_gadget`), but in case of this tutorial, it's preferred to have a static IP address.
+
+To set static networking, you need to add the configuration to `config/config.exs`:
+
+```
+# Statically assign an address
+config :nerves_network, :default,
+  eth0: [
+    ipv4_address_method: :static,
+    ipv4_address: "192.168.88.2",
+    ipv4_subnet_mask: "255.255.255.0",
+    nameservers: ["8.8.8.8", "8.8.4.4"]
+  ]
+ ```
+
+Please note that this config is for a wired connection. If you want to use wireless connection - take a look at [Nerves network documentation](https://github.com/nerves-project/nerves_network#wifi-networking).
+
+Note that you need to use your local network parameters here - in my network there is an unallocated IP`192.168.88.2`, which I am going to use. However, in your case, it may differ.
```

Comment:
```suggestion
Note that you need to use your local network parameters here - in my network there is an unallocated IP `192.168.88.2`, which I am going to use. However, in your case, it may differ.
```